packages/cardano-services/src/util/DbSyncProvider/DbSyncProvider.ts

```diff
@@ -1,16 +1,38 @@
 import { Cardano, CardanoNode, HealthCheckResponse, Provider, ProviderDependencies } from '@cardano-sdk/core';
 import { DB_BLOCKS_BEHIND_TOLERANCE, DB_MAX_SAFE_INTEGER, LedgerTipModel, findLedgerTip } from './util';
+import { DEFAULT_HEALTH_CHECK_CACHE_TTL } from '../../Program/options';
+import { InMemoryCache } from '../../InMemoryCache';
 import { Logger } from 'ts-log';
 import { Pool } from 'pg';
 
+/**
+ * Dedicated DB pools
+ */
+export interface DbPools {
+  /**
+   * Main operational db pool
+   */
+  main?: Pool;
+  /**
+   * Secondary health check db pool
+   */
+  healthCheck?: Pool;
+}
+
 /**
  * Properties that are need to create DbSyncProvider
  */
 export interface DbSyncProviderDependencies extends ProviderDependencies {
   /**
-   * DB connection
+   * Cache engines. Default: InMemoryCache with HealthCheck.cacheTTL as default TTL
+   */
+  cache?: {
```

Comment:
Both `cache` and `healthCheck` should be required.
All the `DbSyncProvider`s should share the same cache instance; if each `DbSyncProvider` instantiate its own cache, we are going to repeat the same query for each of them.

Reply:
426169643da6936df4ff1d85b19c29fc897c9135